From 1757565bea4a6f4f2503ffa0a54b07d436a8fc29 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 6 May 2019 11:57:31 +0200
Subject: [PATCH 226/386] arm64: dts: cn9130: fix pcie addresses

The pcie mem base address configured by firmware is:
- 0xe0000000 for pcie1
- 0xe1000000 for pcie2

Fix the mem base macro so the addresses are aligned to fw settings.

Change-Id: I54e97045505b62587848244b851f38b842cba4f8
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/8772
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index 44826a0a376c..bdd215df7c95 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -34,7 +34,7 @@
 #define CP110_NUM			0
 #define CP110_PCIEx_CPU_IO_BASE(iface)	(0xf8000000 + (iface) * 0x10000)
 #define CP110_PCIEx_CPU_MEM_BASE(iface)	((iface == 0) ? 0xc0000000 : \
-					   (0xe0000000 + (iface) * 0x1000000))
+					 (0xe0000000 + (iface - 1) * 0x1000000))
 #define CP110_PCIEx_BUS_IO_BASE(iface)	(CP110_PCIEx_CPU_IO_BASE(iface))
 #define CP110_PCIEx_BUS_MEM_BASE(iface)	(CP110_PCIEx_CPU_MEM_BASE(iface))
 
-- 
2.17.1

